meta-lxatac-bsp/recipes-kernel/linux/files/patches/0304-ARM-dts-stm32-lxa-tac-Add-support-for-generation-3-d.patch

```diff
@@ -0,0 +1,605 @@
++// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
++/*
++ * Copyright (C) 2020 STMicroelectronics - All Rights Reserved
++ * Copyright (C) 2021 Rouven Czerwinski, Pengutronix
++ * Copyright (C) 2024 Leonard Göhrs, Pengutronix
++ */
++
++/dts-v1/;
++
++#include "stm32mp153.dtsi"
++#include "stm32mp15xc-lxa-tac.dtsi"
++
++/ {
++	model = "Linux Automation Test Automation Controller (TAC) Gen 3";
++	compatible = "lxa,stm32mp153c-tac-gen3", "oct,stm32mp15xx-osd32", "st,stm32mp153";
++
++	backlight: backlight {
++		compatible = "pwm-backlight";
++		power-supply = <&v3v3>;
++
++		brightness-levels = <0 31 63 95 127 159 191 223 255>;
++		default-brightness-level = <7>;
++		pwms = <&led_pwm 3 1000000 0>;
++	};
++
++	reg_iobus_12v: regulator-iobus-12v {
++		compatible = "regulator-fixed";
++		vin-supply = <&reg_12v>;
++		gpio = <&gpioh 13 GPIO_ACTIVE_LOW>;
++		regulator-max-microvolt = <12000000>;
++		regulator-min-microvolt = <12000000>;
++		regulator-name = "12V_IOBUS";
++	};
++
++	led-controller-1 {
++		compatible = "pwm-leds-multicolor";
++
++		multi-led {
++			color = <LED_COLOR_ID_RGB>;
++			function = LED_FUNCTION_STATUS;
++			max-brightness = <65535>;
++
++			led-red {
++				active-low;
++				color = <LED_COLOR_ID_RED>;
++				pwms = <&led_pwm 0 1000000 0>;
++			};
++
++			led-green {
++				active-low;
++				color = <LED_COLOR_ID_GREEN>;
++				pwms = <&led_pwm 2 1000000 0>;
++			};
++
++			led-blue {
++				active-low;
++				color = <LED_COLOR_ID_BLUE>;
++				pwms = <&led_pwm 1 1000000 0>;
++			};
++		};
++	};
++
++	led-controller-2 {
++		compatible = "gpio-leds";
++
++		led-5 {
++			label = "tac:green:iobus";
++			gpios = <&gpiog 1 GPIO_ACTIVE_HIGH>;
++		};
++
++		led-6 {
++			label = "tac:green:can";
++			gpios = <&gpiof 3 GPIO_ACTIVE_HIGH>;
++		};
++
++		led-7 {
++			label = "tac:green:out0";
++			gpios = <&gpiob 8 GPIO_ACTIVE_HIGH>;
++		};
++
++		led-8 {
++			label = "tac:green:out1";
++			gpios = <&gpiog 3 GPIO_ACTIVE_HIGH>;
++		};
++
++		led-9 {
++			label = "tac:green:uarttx";
++			gpios = <&gpiod 3 GPIO_ACTIVE_HIGH>;
++		};
++
++		led-10 {
++			label = "tac:green:uartrx";
++			gpios = <&gpiof 6 GPIO_ACTIVE_HIGH>;
++		};
++
++		led-11 {
++			label = "tac:green:usbh1";
++			gpios = <&gpioc 8 GPIO_ACTIVE_HIGH>;
++		};
++
++		led-12 {
++			label = "tac:green:usbh2";
++			gpios = <&gpiod 6 GPIO_ACTIVE_HIGH>;
++		};
++
++		led-13 {
++			label = "tac:green:usbh3";
++			gpios = <&gpiob 9 GPIO_ACTIVE_HIGH>;
++		};
++
++		led-14 {
++			label = "tac:green:usbg";
++			gpios = <&gpiod 14 GPIO_ACTIVE_HIGH>;
++			linux,default-trigger = "usb-gadget";
++		};
++
++		led-15 {
++			label = "tac:green:dutpwr";
++			gpios = <&gpioa 15 GPIO_ACTIVE_HIGH>;
++		};
++	};
++};
++
++&adc {
++	pinctrl-names = "default";
++	pinctrl-0 = <&adc1_ain_pins_b>;
++	vdd-supply = <&vdd>;
++	vdda-supply = <&vdda>;
++	vref-supply = <&vrefbuf>;
++	status = "okay";
++
++	adc1: adc@0 {
++		st,adc-channels = <2 5 9 10 13 14 15 18>;
++		st,min-sample-time-nsecs = <5000>;
```

Comment:
Is my understanding correct that this equals 200k samples/second? Isn't this a bit much?

Reply:
That's the maximum frequency the ADC may run at. The actual sample frequency used is more or less independent from this value. In the `tacd` we use a much lower sample rate of [80Hz](https://github.com/linux-automation/tacd/blob/002ba0887a9a6ba581608653c3966f0efe49cb46/src/adc/iio/hardware.rs#L440).